File: Professional/EndUserDesigner/C#/EndUserDesigner/ExportForm.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;
using GrapeCity.ActiveReports.Export.Html.Page;
using GrapeCity.ActiveReports.Export.Text.Page;
using GrapeCity.ActiveReports.Extensibility.Rendering.IO;
using GrapeCity.ActiveReports.Rendering.IO;
using System.IO;
using GrapeCity.ActiveReports.Export.Image.Page;
using GrapeCity.ActiveReports.Export.Pdf.Page;
using GrapeCity.ActiveReports.Export.Rdf;
using GrapeCity.ActiveReports.Export.Word.Page;
using GrapeCity.ActiveReports.Export.Xml.Page;
using GrapeCity.ActiveReports.Export.Excel.Page;
using GrapeCity.ActiveReports.Design;
using GrapeCity.ActiveReports.Export.Excel.Section;
using System.Xml;
namespace GrapeCity.ActiveReports.Samples.EndUserDesigner
{
	public partial class ExportForm : Form
	{
		readonly DesignerReportType _reportType;
		readonly object _report;
		private readonly Viewer.Win.Viewer _reportViewer;
		readonly SectionReport _sectionReport = new SectionReport();
		public ExportForm(DesignerReportType reportType, object report, Viewer.Win.Viewer reportViewer)
		{
			_reportType = reportType;
			_report = report;
			_reportViewer = reportViewer;
			InitializeComponent();
		}
		void AddPageReportExportTypes()
		{
			List<PageExport> pageExports = new List<PageExport>();
			bool isFpl = _reportType == DesignerReportType.Page;
			var imageRenderingExtension = new ImageRenderingExtension();
			var wordRenderingExtension = new WordRenderingExtension();
			var htmlRenderingExtension = new HtmlRenderingExtension();
			var pdfRenderingExtension = new PdfRenderingExtension();
			var xmlRenderingExtension = new XmlRenderingExtension();
			var excelRenderingExtension = new ExcelRenderingExtension();
			var csvRenderingExtension = new CsvRenderingExtension();
			pageExports.Add(new PageExport
			{
				DisplayName = Properties.Resources.PageCsvString,
				DefaultFileName = Properties.Resources.CsvFileName,
				Settings = csvRenderingExtension.GetSupportedSettings(isFpl),
				Filter = Properties.Resources.CsvFilter,
				RenderingExtension = csvRenderingExtension,
			});
			pageExports.Add(new PageExport
			{
				DisplayName = Properties.Resources.PageImageString,
				DefaultFileName = Properties.Resources.ImageFileName,
				Settings = imageRenderingExtension.GetSupportedSettings(isFpl),
				Filter = Properties.Resources.ImageFilter,
				RenderingExtension = imageRenderingExtension,
			});
			pageExports.Add(new PageExport
			{
				DisplayName = Properties.Resources.PageWordString,
				DefaultFileName = Properties.Resources.WordFileName,
				Settings = wordRenderingExtension.GetSupportedSettings(isFpl),
				Filter = Properties.Resources.WordFilter,
				RenderingExtension = wordRenderingExtension,
			});
			pageExports.Add(new PageExport
			{
				DisplayName = Properties.Resources.PageHTMLString,
				DefaultFileName = Properties.Resources.HTMLFileName,
				Settings = htmlRenderingExtension.GetSupportedSettings(isFpl),
				Filter = Properties.Resources.HTMLFilter,
				RenderingExtension = htmlRenderingExtension,
			});
			pageExports.Add(new PageExport
			{
				DisplayName = Properties.Resources.PagePDFString,
				DefaultFileName = Properties.Resources.PDFFileName,
				Settings = pdfRenderingExtension.GetSupportedSettings(isFpl),
				Filter = Properties.Resources.PDFFilter,
				RenderingExtension = pdfRenderingExtension,
			});
			pageExports.Add(new PageExport
			{
				DisplayName = Properties.Resources.PageRDFString,
				DefaultFileName = Properties.Resources.RDFFileName,
				Settings = new Export.Rdf.Settings(),
				Filter = Properties.Resources.RdfFilter,
				RenderingExtension = new RdfRenderingExtension(),
			});
			pageExports.Add(new PageExport
			{
				DisplayName = Properties.Resources.PageXMLString,
				DefaultFileName = Properties.Resources.XMLFileName,
				Settings = xmlRenderingExtension.GetSupportedSettings(isFpl),
				Filter = Properties.Resources.XmlFilter,
				RenderingExtension = xmlRenderingExtension,
			});
			pageExports.Add(new PageExport
			{
				DisplayName = (_reportType == DesignerReportType.Rdl) ? Properties.Resources.PageExcelRE : Properties.Resources.PageExcelString,
				DefaultFileName = Properties.Resources.ExcelFileName,
				Settings = excelRenderingExtension.GetSupportedSettings(isFpl),
				Filter = Properties.Resources.ExcelFilter,
				RenderingExtension = excelRenderingExtension,
			});
			cmbExportFormat.DataSource = pageExports;
			cmbExportFormat.DisplayMember = "DisplayName";
			cmbExportFormat.SelectedIndex = 0;
		}
		void AddSectionReportExportTypes()
		{
			List<SectionExport> sectionExports = new List<SectionExport>();
			sectionExports.Add(new SectionExport
			{
				DisplayName = Properties.Resources.SectionHTMLString,
				Filter = Properties.Resources.HTMLFilter,
				DefaultFileName = Properties.Resources.HTMLFileName,
				Settings = new Export.Html.Section.HtmlExport()
			});
			sectionExports.Add(new SectionExport
			{
				DisplayName = Properties.Resources.SectionPDFString,
				Filter = Properties.Resources.PDFFilter,
				DefaultFileName = Properties.Resources.PDFFileName,
				Settings = new Export.Pdf.Section.PdfExport()
			});
			sectionExports.Add(new SectionExport
			{
				DisplayName = Properties.Resources.SectionRTFString,
				Filter = Properties.Resources.RtfFilter,
				DefaultFileName = Properties.Resources.RtfFileName,
				Settings = new Export.Word.Section.RtfExport()
			});
			sectionExports.Add(new SectionExport
			{
				DisplayName = Properties.Resources.SectionTIFFString,
				Filter = Properties.Resources.TiffFilter,
				DefaultFileName = Properties.Resources.TiffFileName,
				Settings = new Export.Image.Tiff.Section.TiffExport()
			});
			sectionExports.Add(new SectionExport
			{
				DisplayName = Properties.Resources.SectionPlainTextString,
				Filter = Properties.Resources.TextFilter,
				DefaultFileName = Properties.Resources.TextFileName,
				Settings = new Export.Xml.Section.TextExport()
			});
			sectionExports.Add(new SectionExport
			{
				DisplayName = Properties.Resources.SectionExcelString,
				Filter = Properties.Resources.ExcelFilter,
				DefaultFileName = Properties.Resources.ExcelFileName,
				Settings = new XlsExport()
			});
			cmbExportFormat.DataSource = sectionExports;
			cmbExportFormat.DisplayMember = "DisplayName";
			cmbExportFormat.SelectedIndex = 0;
		}
		private void ExportForm_Load(object sender, EventArgs e)
		{
			if (_reportType == DesignerReportType.Section)
			{
				AddSectionReportExportTypes();
			}
			if ((_reportType == DesignerReportType.Page) || (_reportType == DesignerReportType.Rdl))
			{
				AddPageReportExportTypes();
			}
		}
		private void cmbExportFormat_SelectedIndexChanged(object sender, EventArgs e)
		{
			if ((_reportType == DesignerReportType.Page) || (_reportType == DesignerReportType.Rdl))
			{
				exportPropertyGrid.SelectedObject = ((PageExport)cmbExportFormat.SelectedItem).Settings;
			}
			if (_reportType == DesignerReportType.Section)
			{
				exportPropertyGrid.SelectedObject = ((SectionExport)cmbExportFormat.SelectedItem).Settings;
			}
		}
		private void btnCancel_Click(object sender, EventArgs e)
		{
			Hide();
		}
		private void btnOk_Click(object sender, EventArgs e)
		{
			switch (_reportType)
			{
				case DesignerReportType.Section:
					exportSaveFileDialog.Filter = ((SectionExport)cmbExportFormat.SelectedItem).Filter;
					exportSaveFileDialog.FileName = ((SectionExport)cmbExportFormat.SelectedItem).DefaultFileName;
					if (exportPropertyGrid.SelectedObject is XlsExport)
					{
						switch (((XlsExport)exportPropertyGrid.SelectedObject).FileFormat)
						{
							case Export.Excel.Section.FileFormat.Xlsx:
								exportSaveFileDialog.FileName = Properties.Resources.Excel2007FileName;
								exportSaveFileDialog.Filter = Properties.Resources.Excel2007Filter;
								break;
						}
					}
					if (exportSaveFileDialog.ShowDialog() == DialogResult.OK)
					{
						RunSectionReport();
						if (File.Exists(exportSaveFileDialog.FileName))
						{
							File.Delete(exportSaveFileDialog.FileName);
						}
						((SectionExport)cmbExportFormat.SelectedItem).Export(_sectionReport.Document, exportSaveFileDialog.FileName);
						MessageBox.Show(((SectionExport)cmbExportFormat.SelectedItem).DisplayName + " " + Properties.Resources.ExportCompleteString);
						Hide();
					}
					break;
				case DesignerReportType.Page:
				case DesignerReportType.Rdl:
					exportSaveFileDialog.AddExtension = false;
					exportSaveFileDialog.Filter = ((PageExport)cmbExportFormat.SelectedItem).Filter;
					exportSaveFileDialog.FileName = ((PageExport)cmbExportFormat.SelectedItem).DefaultFileName;
					if (exportPropertyGrid.SelectedObject is Export.Image.Page.Settings)
					{
						Export.Image.Page.Settings _settings = (Export.Image.Page.Settings)exportPropertyGrid.SelectedObject;
						exportSaveFileDialog.FileName = ((PageExport)cmbExportFormat.SelectedItem).DefaultFileName + "." + _settings.ImageType.ToString().ToLower();
					}
					if (exportPropertyGrid.SelectedObject is Export.Word.Page.Settings &&
						((Export.Word.Page.Settings)exportPropertyGrid.SelectedObject).FileFormat ==
							Export.Word.Page.FileFormat.OOXML)
					{
						exportSaveFileDialog.FileName = Properties.Resources.Word2007FileName;
						exportSaveFileDialog.Filter = Properties.Resources.Word2007Filter;
					}
					if (exportPropertyGrid.SelectedObject is ExcelRenderingExtensionSettings &&
						((ExcelRenderingExtensionSettings) exportPropertyGrid.SelectedObject).FileFormat ==
							Export.Excel.Page.FileFormat.Xlsx)
					{
						exportSaveFileDialog.FileName = Properties.Resources.Excel2007FileName;
						exportSaveFileDialog.Filter = Properties.Resources.Excel2007Filter;
					}
					if (exportSaveFileDialog.ShowDialog() == DialogResult.OK)
					{
						if (File.Exists(exportSaveFileDialog.FileName))
						{
							File.Delete(exportSaveFileDialog.FileName);
						}
						PageExport tempQualifier = ((PageExport)cmbExportFormat.SelectedItem);
						string filePath = exportSaveFileDialog.FileName;
						StreamProvider StreamProvider = new FileStreamProvider(new DirectoryInfo(Path.GetDirectoryName(filePath)), Path.GetFileNameWithoutExtension(filePath));
						_reportViewer.Render(tempQualifier.RenderingExtension, StreamProvider, tempQualifier.Settings.GetSettings());
						MessageBox.Show(((PageExport)cmbExportFormat.SelectedItem).DisplayName + " " + Properties.Resources.ExportCompleteString);
						Hide();
					}
					break;
			}
		}
		void RunSectionReport()
		{
			MemoryStream memoryStream = new MemoryStream();
			memoryStream.Position = 0;
			using (XmlWriter writer = XmlWriter.Create(memoryStream))
			{
				((SectionReport)_report).SaveLayout(writer);
			}
			memoryStream.Position = 0;
			using (XmlReader reader = XmlReader.Create(memoryStream))
			{
				_sectionReport.LoadLayout(reader);
			}
			_sectionReport.Run();
		}
	}
}

File: Professional/EndUserDesigner/VB.NET/EndUserDesigner/ExportForm.vb
Imports System.Collections.Generic
Imports System.Windows.Forms
Imports GrapeCity.ActiveReports.Export.Html.Page
Imports GrapeCity.ActiveReports.Rendering.IO
Imports System.IO
Imports GrapeCity.ActiveReports.Export.Image.Page
Imports GrapeCity.ActiveReports.Export.Pdf.Page
Imports GrapeCity.ActiveReports.Export.Rdf
Imports GrapeCity.ActiveReports.Export.Word.Page
Imports GrapeCity.ActiveReports.Export.Xml.Page
Imports GrapeCity.ActiveReports.Export.Excel.Page
Imports GrapeCity.ActiveReports.Design
Imports GrapeCity.ActiveReports.Export.Excel.Section
Imports System.Xml
Imports GrapeCity.ActiveReports.Extensibility.Rendering.IO
Imports GrapeCity.ActiveReports.Export.Text.Page
Public Class ExportForm
	Dim _reportType As DesignerReportType
	Dim _report As Object
	Private ReadOnly _reportViewer As Viewer.Win.Viewer
	Dim _sectionReport As New SectionReport()
	Public Sub New(ByVal ReportType As DesignerReportType, ByVal Report As Object, ByVal reportViewer As GrapeCity.ActiveReports.Viewer.Win.Viewer)
		_reportType = ReportType
		_report = Report
		_reportViewer = reportViewer
		InitializeComponent()
	End Sub
	Private Sub AddPageReportExportTypes()
		Dim pageExports As New List(Of PageExport)()
		Dim isFpl As Boolean = _reportType = DesignerReportType.Page
		Dim imageRenderingExtension As ImageRenderingExtension = New ImageRenderingExtension()
		Dim wordRenderingExtension As WordRenderingExtension = New WordRenderingExtension()
		Dim htmlRenderingExtension As HtmlRenderingExtension = New HtmlRenderingExtension()
		Dim pdfRenderingExtension As PdfRenderingExtension = New PdfRenderingExtension()
		Dim xmlRenderingExtension As XmlRenderingExtension = New XmlRenderingExtension()
		Dim excelRenderingExtension As ExcelRenderingExtension = New ExcelRenderingExtension()
        Dim csvRenderingExtension As CsvRenderingExtension = New CsvRenderingExtension()
		pageExports.Add(New PageExport() With { _
			.DisplayName = My.Resources.Resource.PageCsvString, _
			.DefaultFileName = My.Resources.Resource.CsvFileName, _
			.Settings = csvRenderingExtension.GetSupportedSettings(isFpl), _
			.Filter = My.Resources.Resource.CsvFilter, _
			.RenderingExtension = csvRenderingExtension _
		})
		pageExports.Add(New PageExport() With { _
			.DisplayName = My.Resources.Resource.PageImageString, _
			.DefaultFileName = My.Resources.Resource.ImageFileName, _
			.Settings = imageRenderingExtension.GetSupportedSettings(isFpl), _
			.Filter = My.Resources.Resource.ImageFilter, _
			.RenderingExtension = imageRenderingExtension _
		})
		pageExports.Add(New PageExport() With { _
			.DisplayName = My.Resources.Resource.PageWordString, _
			.DefaultFileName = My.Resources.Resource.WordFileName, _
			.Settings = wordRenderingExtension.GetSupportedSettings(isFpl), _
			.Filter = My.Resources.Resource.WordFilter, _
			.RenderingExtension = wordRenderingExtension _
		})
		pageExports.Add(New PageExport() With { _
			.DisplayName = My.Resources.Resource.PageHTMLString, _
			.DefaultFileName = My.Resources.Resource.HTMLFileName, _
			.Settings = htmlRenderingExtension.GetSupportedSettings(isFpl), _
			.Filter = My.Resources.Resource.HTMLFilter, _
			.RenderingExtension = htmlRenderingExtension _
		})
		pageExports.Add(New PageExport() With { _
			.DisplayName = My.Resources.Resource.PagePDFString, _
			.DefaultFileName = My.Resources.Resource.PDFFileName, _
			.Settings = pdfRenderingExtension.GetSupportedSettings(isFpl), _
			.Filter = My.Resources.Resource.PDFFilter, _
			.RenderingExtension = pdfRenderingExtension _
		})
		pageExports.Add(New PageExport() With { _
			.DisplayName = My.Resources.Resource.PageRDFString, _
			.DefaultFileName = My.Resources.Resource.RDFFileName, _
			.Settings = New Export.Rdf.Settings(), _
			.Filter = My.Resources.Resource.RdfFilter, _
			.RenderingExtension = New RdfRenderingExtension() _
		})
		pageExports.Add(New PageExport() With { _
			.DisplayName = My.Resources.Resource.PageXMLString, _
			.DefaultFileName = My.Resources.Resource.XMLFileName, _
			.Settings = xmlRenderingExtension.GetSupportedSettings(isFpl), _
			.Filter = My.Resources.Resource.XmlFilter, _
			.RenderingExtension = xmlRenderingExtension _
		})
		pageExports.Add(New PageExport() With { _
			.DisplayName = If((_reportType = DesignerReportType.Rdl), My.Resources.Resource.PageExcelRE, My.Resources.Resource.PageExcelString), _
			.DefaultFileName = My.Resources.Resource.ExcelFileName, _
			.Settings = excelRenderingExtension.GetSupportedSettings(isFpl), _
			.Filter = My.Resources.Resource.ExcelFilter, _
			.RenderingExtension = excelRenderingExtension _
		})
		cmbExportFormat.DataSource = pageExports
		cmbExportFormat.DisplayMember = "DisplayName"
		cmbExportFormat.SelectedIndex = 0
	End Sub
	Private Sub AddSectionReportExportTypes()
		Dim sectionExports As New List(Of SectionExport)()
		sectionExports.Add(New SectionExport() With { _
			.DisplayName = My.Resources.Resource.SectionHTMLString, _
			.Filter = My.Resources.Resource.HTMLFilter, _
			.DefaultFileName = My.Resources.Resource.HTMLFileName, _
			.Settings = New Export.Html.Section.HtmlExport() _
		})
		sectionExports.Add(New SectionExport() With { _
			.DisplayName = My.Resources.Resource.SectionPDFString, _
			.Filter = My.Resources.Resource.PDFFilter, _
			.DefaultFileName = My.Resources.Resource.PDFFileName, _
			.Settings = New Export.Pdf.Section.PdfExport() _
		})
		sectionExports.Add(New SectionExport() With { _
			.DisplayName = My.Resources.Resource.SectionRTFString, _
			.Filter = My.Resources.Resource.RtfFilter, _
			.DefaultFileName = My.Resources.Resource.RtfFileName, _
			.Settings = New Export.Word.Section.RtfExport() _
		})
		sectionExports.Add(New SectionExport() With { _
			.DisplayName = My.Resources.Resource.SectionTIFFString, _
			.Filter = My.Resources.Resource.TiffFilter, _
			.DefaultFileName = My.Resources.Resource.TiffFileName, _
			.Settings = New Export.Image.Tiff.Section.TiffExport() _
		})
		sectionExports.Add(New SectionExport() With { _
			.DisplayName = My.Resources.Resource.SectionPlainTextString, _
			.Filter = My.Resources.Resource.TextFilter, _
			.DefaultFileName = My.Resources.Resource.TextFileName, _
			.Settings = New Export.Xml.Section.TextExport() _
		})
		sectionExports.Add(New SectionExport() With { _
			.DisplayName = My.Resources.Resource.SectionExcelString, _
			.Filter = My.Resources.Resource.ExcelFilter, _
			.DefaultFileName = My.Resources.Resource.ExcelFileName, _
			.Settings = New Export.Excel.Section.XlsExport() _
		})
		cmbExportFormat.DataSource = sectionExports
		cmbExportFormat.DisplayMember = "DisplayName"
		cmbExportFormat.SelectedIndex = 0
	End Sub
	Private Sub ExportForm_Load(ByVal sender As System.Object, ByVal e As EventArgs) Handles MyBase.Load
		If _reportType = DesignerReportType.Section Then
			AddSectionReportExportTypes()
		End If
		If (_reportType = DesignerReportType.Page) OrElse (_reportType = DesignerReportType.Rdl) Then
			AddPageReportExportTypes()
		End If
	End Sub
	Private Sub cmbExportFormat_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As EventArgs) Handles cmbExportFormat.SelectedIndexChanged
		If (_reportType = DesignerReportType.Page) OrElse (_reportType = DesignerReportType.Rdl) Then
			exportPropertyGrid.SelectedObject = DirectCast(cmbExportFormat.SelectedItem, PageExport).Settings
		End If
		If _reportType = DesignerReportType.Section Then
			exportPropertyGrid.SelectedObject = DirectCast(cmbExportFormat.SelectedItem, SectionExport).Settings
		End If
	End Sub
	Private Sub btnCancel_Click(ByVal sender As System.Object, ByVal e As EventArgs) Handles btnCancel.Click
		Hide()
	End Sub
	Private Sub btnOk_Click(ByVal sender As System.Object, ByVal e As EventArgs) Handles btnOk.Click
		Select Case _reportType
			Case DesignerReportType.Section
				exportSaveFileDialog.Filter = DirectCast(cmbExportFormat.SelectedItem, SectionExport).Filter
				exportSaveFileDialog.FileName = DirectCast(cmbExportFormat.SelectedItem, SectionExport).DefaultFileName
				If TypeOf exportPropertyGrid.SelectedObject Is XlsExport Then
					Select Case DirectCast(exportPropertyGrid.SelectedObject, XlsExport).FileFormat
						Case Export.Excel.Section.FileFormat.Xlsx
							exportSaveFileDialog.FileName = My.Resources.Resource.Excel2007FileName
							exportSaveFileDialog.Filter = My.Resources.Resource.Excel2007Filter
							Exit Select
					End Select
				End If
				If exportSaveFileDialog.ShowDialog() = DialogResult.OK Then
					RunSectionReport()
					If File.Exists(exportSaveFileDialog.FileName) Then
						File.Delete(exportSaveFileDialog.FileName)
					End If
					DirectCast(cmbExportFormat.SelectedItem, SectionExport).Export(_sectionReport.Document, exportSaveFileDialog.FileName)
					MessageBox.Show(DirectCast(cmbExportFormat.SelectedItem, SectionExport).DisplayName + " " + My.Resources.Resource.ExportCompleteString)
					Hide()
				End If
				Exit Select
			Case DesignerReportType.Page, DesignerReportType.Rdl
				exportSaveFileDialog.AddExtension = False
				exportSaveFileDialog.Filter = DirectCast(cmbExportFormat.SelectedItem, PageExport).Filter
				exportSaveFileDialog.FileName = DirectCast(cmbExportFormat.SelectedItem, PageExport).DefaultFileName
				If TypeOf exportPropertyGrid.SelectedObject Is Export.Image.Page.Settings Then
					Dim settings As Export.Image.Page.Settings = DirectCast(exportPropertyGrid.SelectedObject, Export.Image.Page.Settings)
					exportSaveFileDialog.FileName = DirectCast(cmbExportFormat.SelectedItem, PageExport).DefaultFileName + "." & settings.ImageType.ToString().ToLower()
				End If
				If TypeOf exportPropertyGrid.SelectedObject Is Export.Word.Page.Settings Then
					Select Case DirectCast(exportPropertyGrid.SelectedObject, Export.Word.Page.Settings).FileFormat
						Case Export.Word.Page.FileFormat.OOXML
							exportSaveFileDialog.FileName = My.Resources.Resource.Word2007FileName
							exportSaveFileDialog.Filter = My.Resources.Resource.Word2007Filter
							Exit Select
					End Select
				End If
				If TypeOf exportPropertyGrid.SelectedObject Is ExcelRenderingExtensionSettings Then
					Select Case DirectCast(exportPropertyGrid.SelectedObject, ExcelRenderingExtensionSettings).FileFormat
						Case Export.Excel.Page.FileFormat.Xlsx
							exportSaveFileDialog.FileName = My.Resources.Resource.Excel2007FileName
							exportSaveFileDialog.Filter = My.Resources.Resource.Excel2007Filter
							Exit Select
					End Select
				End If
				If exportSaveFileDialog.ShowDialog() = DialogResult.OK Then
					If File.Exists(exportSaveFileDialog.FileName) Then
						File.Delete(exportSaveFileDialog.FileName)
					End If
					Dim export As PageExport = DirectCast(cmbExportFormat.SelectedItem, PageExport)
					Dim StreamProvider As StreamProvider = New FileStreamProvider(New DirectoryInfo(Path.GetDirectoryName(exportSaveFileDialog.FileName)), Path.GetFileNameWithoutExtension(exportSaveFileDialog.FileName))
					_reportViewer.Render(export.RenderingExtension, StreamProvider, export.Settings.GetSettings())
					MessageBox.Show(DirectCast(cmbExportFormat.SelectedItem, PageExport).DisplayName + " " + My.Resources.Resource.ExportCompleteString)
					Hide()
				End If
				Exit Select
		End Select
	End Sub
	Private Sub RunSectionReport()
		Dim memoryStream As New MemoryStream()
		memoryStream.Position = 0
		Using writer As XmlWriter = XmlWriter.Create(memoryStream)
			DirectCast(_report, SectionReport).SaveLayout(writer)
		End Using
		memoryStream.Position = 0
		Using reader As XmlReader = XmlReader.Create(memoryStream)
			_sectionReport.LoadLayout(reader)
		End Using
		_sectionReport.Run()
	End Sub
End Class
